File: NetLibraryGenerator/NetLibraryGenerator/Model/Results/ChangedMethod.cs
using Newtonsoft.Json;

namespace NetLibraryGenerator.Model.Results;

public class ChangedMethod
{
    public ChangedMethod(string category, string name, string oldSignature, string newSignature)
    {
        Category = category;
        Name = name;
        OldSignature = oldSignature;
        NewSignature = newSignature;
    }

    [JsonProperty("category")]
    public string Category { get; }

    [JsonProperty("name")]
    public string Name { get; }
    
    [JsonProperty("oldSignature")]
    public string OldSignature { get; }
    
    [JsonProperty("newSignature")]
    public string NewSignature { get; }
}

File: DartLibraryGenerator/model/results/changed_method.dart
class ChangedMethod {
    ChangedMethod({
        required this.category,
        required this.name,
        required this.oldSignature,
        required this.newSignature,
    });

    final String category;
    final String name;
    final String oldSignature;
    final String newSignature;

    factory ChangedMethod.fromJson(Map<String, dynamic> json) => ChangedMethod(
        category: json["category"],
        name: json["name"],
        oldSignature: json["oldSignature"],
        newSignature: json["newSignature"],
    );

    Map<String, dynamic> toJson() => {
        "category": category,
        "name": name,
        "oldSignature": oldSignature,
        "newSignature": newSignature,
    };
}